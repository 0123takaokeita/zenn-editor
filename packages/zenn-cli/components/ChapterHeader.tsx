import { Chapter } from "@types";
import { getChapterErrors } from "@utils/validator";
import ContentWrapper from "@components/ContentWrapper";
import ErrorRow from "@components/ErrorRow";

const ArticleHeader: React.FC<{ chapter: Chapter }> = ({ chapter }) => {
  const errorMessages = getChapterErrors(chapter);
  const errorCount = errorMessages?.length;
  return (
    <header className="content-header">
      <ContentWrapper>
        <div style={{ fontSize: "14px" }}>Chapter {chapter.position}</div>

        <h1 className="content-header__title">{chapter.title || "No Title"}</h1>

        {!!errorCount && (
          <div>
            <div className="content-header__error-title">
              {errorCount}件の修正が必要です
            </div>
            {errorMessages.map((errorMessage, index) => (
              <ErrorRow errorMessage={errorMessage} key={`invldmsg${index}`} />
            ))}
          </div>
        )}

        <a href="todo" className="content-header__link" target="_blank">
          Chapterのmdファイルの作成方法はこちら →
        </a>
      </ContentWrapper>
    </header>
  );
};
export default ArticleHeader;
