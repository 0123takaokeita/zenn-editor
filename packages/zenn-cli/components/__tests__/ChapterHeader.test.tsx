import React from "react";
import { createRenderer } from "react-test-renderer/shallow";
import { ChapterHeader } from "@components/ChapterHeader";
import { Chapter } from "@types";

const renderer = createRenderer();
const dummyChapter: Chapter = {
  content: "",
  free: false,
  position: undefined,
  slug: "",
  title: "",
};

describe("<ChapterHeader />", () => {
  it("should render and match the snapshot", () => {
    renderer.render(<ChapterHeader chapter={dummyChapter} />);
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});
