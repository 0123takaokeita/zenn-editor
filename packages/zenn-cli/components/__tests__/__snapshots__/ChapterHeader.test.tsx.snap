// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChapterHeader /> should render and match the snapshot 1`] = `
<header
  className="content-header"
>
  <Unknown>
    <div
      className="content-header__chapter-info"
    />
    <h1
      className="content-header__title"
    >
      No Title
    </h1>
    <div
      className="content-header__row"
    >
      <span
        className="content-header__row-title"
      >
        slug
      </span>
      <span
        className="content-header__row-result"
      >
        
      </span>
    </div>
    <div>
      <div
        className="content-header__error-title"
      >
        2
        件の修正が必要です
      </div>
      <Unknown
        errorMessage={
          Object {
            "isCritical": true,
            "message": "チャプターのslugの値（）が不正です。半角英数字（a-z0-9）とハイフン（-）の1〜50字の組み合わせにしてください",
          }
        }
      />
      <Unknown
        errorMessage={
          Object {
            "isCritical": true,
            "message": "title（タイトル）を入力してください",
          }
        }
      />
    </div>
    <a
      className="content-header__link"
      href=" https://zenn.dev/zenn/articles/zenn-cli-guide#cliで本（book）を管理する"
      target="_blank"
    >
      チャプターのmdファイルの作成方法 →
    </a>
  </Unknown>
</header>
`;
